lib/spack/spack/util/web.py

```diff
@@ -289,7 +350,7 @@ def _curl(curl=None):
         except CommandNotFoundError as exc:
             tty.error(str(exc))
             raise spack.error.FetchError("Missing required curl fetch method")
-    return curl
+    return append_curl_env_for_ssl_certs(curl)
```

Comment:
Minor nitpick that I'm not going to hold up the PR over:
* `append_curl_env_for_ssl_certs` modifies the `curl` that is passed in *and* returns it.
* If I read this code, my inclination is to think that I'm getting a copy of `curl` back, but really I'm modifying the same `curl`.
* I think in general you should choose to do either a functional, immutable approach (return a modified and entirely new `curl`) or use side effects, but not both.  If you just want to append, append and don't return anything. The caller already has a reference to the modified thing. Doing both is confusing.

Reply:
good point. yeah we should return a modified copy.

Reply:
Would you prefer side effect or copy in this instance? My initial inclination is copy, but considering the ramifications of potentially have curl fail if used in another instance, perhaps side effect is better... 🤔